src/org/openjump/core/rasterimage/RasterImageLayer.java

```diff
@@ -1580,4 +1580,31 @@ public String getFilePath() {
     //    this.srsInfo = srs;
     //}
     
+      /**
+	 * Check if the RasteImageLayer is spatially consistent with another:
+	 * <li>both have the same cell size
+	 * <li>both have the same dimension (same width and height)
+	 * <li>both overlap to each other into the geographic space
+	 * 
+	 * @param  RasterImageLayer
+	 * @return true if they are spatially consistent, false if they are not
+	 */
+	public  boolean isSpatiallyConsistentWith(RasterImageLayer rasteimageLayer) {
+		if (this.getMetadata().getOriginalCellSize() != rasteimageLayer.getMetadata().getOriginalCellSize()) {
```

Comment:
I think the 3 first tests (xmin, ymin, cellsize) can be replaced by : 
if (! this.getWholeImageEnvelope().equals(rasterImageLayer.getWholeImageEnvelope()) {
  return false;
}